security/s2a/internal/crypter/counter.go

```diff
@@ -0,0 +1,48 @@
+package crypter
+
+import (
+	"encoding/binary"
+	"errors"
+)
+
+// counterLen is the byte length of the counter.
+const counterLen = 8
+
+// counter is a 64-bit, little-endian counter.
+type counter struct {
+	value         uint64
+	hasOverflowed bool
+}
+
+func newCounter(value uint64) counter {
+	return counter{value: value}
+}
+
+// val returns the current value of the counter as a byte slice.
+func (c *counter) val() ([]byte, error) {
+	if c.hasOverflowed {
+		return nil, errors.New("invalid counter due to overflow")
+	}
+	buf := make([]byte, counterLen)
```

Comment:
Per our offline discussion, this should just

Reply:
Done